core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -616,7 +616,10 @@ ClientStream newSubstream(
           parsedDefaultServiceConfig.getError());
       this.defaultServiceConfig =
           (ManagedChannelServiceConfig) parsedDefaultServiceConfig.getConfig();
-      this.lastServiceConfig = this.defaultServiceConfig;
+      if (this.defaultServiceConfig != null
```

Comment:
Remove this `if`. There was a `if (builder.defaultServiceConfig != null) {` check above and it is okay if `thottle == null`.

Reply:
Yep, I was in doubt if this was actually needed, so I added JIC.
Removing it, thank you!